Update EOS Documentation

riccardopersiani opened the pull request:
* Add `#define ORACLIZE_NETWORK` for the random-datasource
* Edit pricing 
* Update old examples
* Fix grammar errors

Commit:
add #define for the random-datasource, edit pricing, update old examples, fix grammar errors

D-Nice commented on source/includes/_eos.md:
```diff
@@ -33,35 +33,35 @@ Before starting, it is necessary to include the `eos_api.hpp` header file. This
+      [[eosio::action]]
+      void execquery() {
+          print("Sending query to Oraclize...");
+          oraclize_query("URL", "json(https://min-api.cryptocompare.com/data/price?fsym=EOS&tsyms=USD).USD",\
```
just stylistic comment, if you will linebreak one of the parameters, might as well linebreak the rest, including URL, and actually save more space. Right now wrapping will still happen. Also, indentation appears to be 4 spaces, so shouldn't these params linebroken follow that, rather tha 2 space indents.

riccardopersiani replied:
I actually prefer to use half the indentation when the code is line-broken, but I am not sure if this is a stylistic best practice. Anyway I would leave this code like this since highlights the proof declaration: 
![image](https://user-images.githubusercontent.com/10484858/52337096-9df4aa00-29fe-11e9-9c49-2e45dc64a296.png)

D-Nice commented on source/includes/_eos.md:
```diff
@@ -116,39 +116,40 @@ Please note that in order for the future timestamp to be accepted by Oraclize it
-         oraclize_query(10, "WolframAlpha", "random number between 1 and 6");
-         print("Oraclize query was sent, standing by for the answer..");  
-      }
+    wolframrand(name receiver, name code, datastream<const char*> ds) : contract(receiver, code, ds) {}
```
Just random comment, I am guessing this is a constructor, and `{}` at the end denotes it doesn't run any specific logic on deploy?

riccardopersiani replied:
Yeah, it's an empty constructor

D-Nice commented on source/includes/_eos.md:
```diff
@@ -116,39 +116,40 @@ Please note that in order for the future timestamp to be accepted by Oraclize it
 
-EOSIO_ABI(example2, (getrandomnum)(callback))
-
+EOSIO_DISPATCH(wolframrand, (getrandomnum)(callback))
```
is this for defiing exposable/external functions, or does `esio::action` specify that?

riccardopersiani replied:
Previously `EOSIO_ABI` and now `EOSIO_DISPATCH` it's for exposing functions. The callable functions of this contract will be `getrandomnum`, `callback` indeed.

D-Nice committed:
chore(docs): update EOS docs for mainnet launch

-Fixes grammar / spelling where applicable
-improve consistencies
-make clear that mainnet is now available, and remove instances which explicitly said it's not yet available
-fix pricing model and mention that contracts dependent on no fee, will stay no fee, but features dependent on fee will not be available to them, aka, legacy price === legacy features

D-Nice reviewed (approved):
Review is mostly stylistic recommendation, and questions. I can't comment a great deal into the correctness of the code, but I have not seen any obvious issues.

I did fix the documentation in this commit: https://github.com/oraclize/docs/pull/20/commits/00071b50b39a108e5cdc22fc346b90fe5cedd9b6

which is also a part of this PR now

Might be good to get confirmation from @bertani on the pricing changes, and on his approval can be merged

bertani commented on source/includes/_eos.md:
```diff
@@ -487,46 +466,58 @@ some specific functions related to the Oraclize Random Data Source have been add
 ## Pricing
 
-The Oraclize integration with EOS is currently available on the **[EOSIO Public "Jungle" Testnet](http://jungle.cryptolions.io/)** only and Oraclize is currently charging no fee there. Our standard [pricing table](#pricing) will apply shortly (`EOS` tokens will be charged), before the launch on the EOSIO Mainnet. The same pricing logic will be applied on testnets too (regardless of their worthles ...
+The Oraclize integration with EOS is currently available on the EOS Mainnet and on the **[EOSIO Public "Jungle" Testnet](http://jungle.cryptolions.io/)**; Oraclize is currently charging **no fee**. Our standard [pricing table](#pricing) may apply (`EOS` tokens will be charged), later on the EOSIO Mainnet. The same pricing logic will be applied on testnets too (regardless of their worthless nature ...
```
*The same pricing logic would be applied on testnets too

bertani commented on source/includes/_eos.md:
```diff
@@ -487,46 +464,62 @@ some specific functions related to the Oraclize Random Data Source have been add
 
+#### Specify The Network Context
+
+It is **mandatory** for the developer to define the _network context_ in which the smart contract, using the Random Data Source,
```
I wouldn't say that it is mandatory as it is not the case.. normal queries would currently work also if the wrong/no chain is specified, however onchain proof verifications would fail. Maybe it is better to say that it is **highly recommended** instead - and remove the "using the random ds" specifier, as the api may require to know the context for other functionalities in the future too

Commit:
fix example style

Commit:
Merge branch 'update-eos-mainnet-docs' of github.com:oraclize/docs into update-eos-mainnet-docs

riccardopersiani committed:
minor changes

Commit:
minor changes

riccardopersiani committed:
approve proposed change

riccardopersiani committed:
Merge branch 'update-eos-mainnet-docs' of github.com:oraclize/docs into update-eos-mainnet-docs

riccardopersiani commented:
I am rebasing and merging
